File: code/water-meter-image-cut.js
require('request');
const http = require('http');
const url = require('url');
const imageCut = require("./lib/lib_image_cut");
const fs = require('fs');

var abfrage = function(req, res) {
    var q = url.parse(req.url, true).query;
    var filename = q.url;

    if (filename)
    {
        console.log(filename);
        var file_download = fs.createWriteStream("./image_tmp/original.jpg");
        http.get(filename, (response) => {response.pipe(file_download);});

        file_download.on('finish', function(){
            file_download.end();

            imageCut.cutImage('./image_tmp', 'original.jpg').then(result => {
                res.writeHead(200, {'Content-Type': 'text/html'});
                var txt = "";
                txt += 'Original: <p><img src=/image_tmp/original.jpg></img><p>';
                txt += 'Rotate: <p><img src=/image_tmp/rot.jpg></img><p>';
                txt += 'Align: <p><img src=/image_tmp/alg.jpg></img><p>';
                txt += 'Counter: <p>'
                for (i = 0; i < result[1].length; ++i)
                {
                    name_zw = result[1][i].substring(1, result[1][i].length);
                    txt += '<img src='+  name_zw + '></img>      ';
                }
                txt += '<p>';
                txt += 'Analog Meter: <p>'
                for (i = 0; i < result[0].length; ++i)
                {
                    name_zw = result[0][i].substring(1, result[0][i].length);
                    txt += '<img src='+  name_zw + '></img>      ';
                }
                txt += '<p>';
                res.end(txt);}
                )
        });
    }

    if (req.url.indexOf('/image_tmp/') !== -1)
    {
        var s = fs.createReadStream("." + req.url);
        s.on('open', function () {
            res.setHeader('Content-Type', 'image/jpeg');
            s.pipe(res);
        });
    }
}

imageCut.init();

http.createServer(abfrage).listen(3000);

File: code/water-meter-image-cut.py
from http.server import HTTPServer, BaseHTTPRequestHandler
from urllib import parse
import urllib.request
import socketserver

import lib.CutImageClass
import cv2


CutImage = lib.CutImageClass.CutImage()

class SimpleHTTPRequestHandler(BaseHTTPRequestHandler):
    
    def do_GET(self):
        global CutImage
        if "/image_tmp/" in self.path:
            self.send_response(200)
            self.send_header('Content-type', 'image/jpeg')
            with open('.'+self.path, 'rb') as file: 
                self.wfile.write(file.read()) # Read the file and send the contents 
        if "url=" in self.path:
            url = parse.parse_qs(parse.urlparse(self.path).query)['url'][0]
            urllib.request.urlretrieve(url, './image_tmp/original.jpg')

            result = CutImage.Cut('./image_tmp/original.jpg')

            txt = 'Original: <p><img src=/image_tmp/original.jpg></img><p>'
            txt = txt + 'Rotate: <p><img src=/image_tmp/rot.jpg></img><p>'
            txt = txt + '<p>Aligned Image: <p><img src=/image_tmp/alg.jpg></img><p>'
            txt = txt + 'Digital Counter: <p>'
            for i in range(len(result[1])):
                txt = txt + '<img src=/image_tmp/'+  str(result[1][i][0]) + '.jpg></img>'
            txt = txt + '<p>'
            txt = txt + 'Analog Meter: <p>'
            for i in range(len(result[0])):
                txt += '<img src=/image_tmp/'+  str(result[0][i][0]) + '.jpg></img>'
            txt = txt + '<p>'

            self.send_response(200)
            self.send_header('Content-type', 'text/html')
            self.end_headers()
            self.wfile.write(bytes(txt, 'UTF-8'))

PORT = 3000

with socketserver.TCPServer(("", PORT), SimpleHTTPRequestHandler) as httpd:
    print("serving at port", PORT)
    httpd.serve_forever()